adagrams/game.py

```diff
@@ -1,11 +1,105 @@
+import random
+from collections import defaultdict
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
 def draw_letters():
-    pass
+    letter_list = []
+    letter_freq = defaultdict(int)
+
+    while len(letter_list) < 10:
+        letter = random.choice(list(LETTER_POOL.keys()))
+        if letter_freq[letter] < LETTER_POOL[letter]:
+            letter_list.append(letter)
+            if letter in letter_freq:
+                letter_freq[letter] += 1
+
+    return letter_list
 
 def uses_available_letters(word, letter_bank):
-    pass
+    is_anagram = True
+    copy_letter_bank = letter_bank[:]
+    word = word.upper()
+
+    for letter in word:
+        if letter not in copy_letter_bank:
+            is_anagram = False
+        else:
+            copy_letter_bank.remove(letter)
+
+    return is_anagram
 
 def score_word(word):
-    pass
+    score = 0
+    word = word.upper()
+    score_chart = {
+        1: ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"],
+        2: ["D", "G"],
+        3: ["B", "C", "M", "P"],
+        4: ["F", "H", "V", "W", "Y"],
+        5: ["K"],
+        8: ["J", "X"],
+        10: ["Q", "Z"]
+    }
```

Comment:
score_chart_dict can be renamed as SCORE_CHART_DICT since it's a constant.

A thought here about the structure of score_chart_dict: I do think that having the  score be the key and having the list of letters as the value makes this data structure look more concise because it takes up fewer lines of code. 

However, when you iterate through the dict on line 73 and then check `if letter in score_chart[key]` on line 74, you already have 2 nested for loops (line 72 and 73) and then the `in` operator checking a list. 

So your worse case scenario is that you have to iterate through every letter in word, iterate through every key/value pair in score_chart, and then iterate through ever letter in score_chart[key] (which is a list) -- quickly growing your time complexity. 

If score_chart_dict was a dict with 26 key value pairs where each letter was the key and each value was the score, then you could leverage a dictionary's constant look up time.